shell/models/__tests__/workload.test.ts

```diff
@@ -0,0 +1,60 @@
+import Workload from '@shell/models/workload.js';
+
+describe('class: Workload', () => {
+  describe('given custom workload keys', () => {
+    const customContainer = {
+      __clone:  'whatever',
+      __active: 'whatever',
+      __init:   'whatever',
+      _init:    'whatever',
+      error:    'whatever',
+    };
+    const customWorkload = {
+      __rehydrate: 'whatever',
+      __clone:     'whatever',
+      spec:        {
+        template: {
+          spec: {
+            containers:     [customContainer],
+            initContainers: [customContainer],
+          }
+        }
+      }
+    };
+
+    it('should keep internal keys', () => {
+      const workload = new Workload(customWorkload, {
+        getters:     { schemaFor: () => ({ linkFor: jest.fn() }) },
+        dispatch:    jest.fn(),
+        rootGetters: { 'i18n/t': jest.fn() },
+      });
+
+      expect({ ...workload }).toStrictEqual(customWorkload);
+    });
+
+    describe('method: save', () => {
+    /**
+     * DISCLAIMER ***************************************************************************************
+     * Logs are prevented to avoid polluting the test output.
+     ****************************************************************************************************
+    */
+      // eslint-disable-next-line jest/no-hooks
+      beforeEach(() => {
+        jest.spyOn(console, 'warn').mockImplementation(() => { });
+      });
+
+      it('should remove all the internal keys', async() => {
+        const workload = new Workload(customWorkload, {
+          getters:     { schemaFor: () => ({ linkFor: jest.fn() }) },
+          dispatch:    jest.fn(),
+          rootGetters: { 'i18n/t': jest.fn() },
+        });
+        const expectation = { spec: { template: { spec: { containers: [{}], initContainers: [{}] } } } };
+
+        await workload.save();
+
+        expect({ ...workload }).toStrictEqual(expectation);
```

Comment:
this test passes though it shouldn't.

the original workload should remain unchanged whilst the payload of the http request kicked off by save should be cleaned

Reply:
What's your suggestion and what specifically would you correct to make it more valid since it's actually passing?

Reply:
the test is a false positive. 

`expect({ ...workload }).toStrictEqual(expectation);` should not be true. there's a bug in code which is making it true.

`workloads` should remain unchanged.

in code `workload.save` kicks off a http request with a payload that should be clean (without properties). it's that which should equal the `expectation`.

i wouldn't worry about these though, i've assigned the original issue to me and will take on the changes required

Reply:
Oh ok so the implementation of the code must be totally wrong then, because we are updating the class and the test is stating that this is correct, but is not our original requirement.

It should be updated also the test to check the request though, no idea where 😅 

We still need a unit test for this.

Reply:
unit tests updated